llvmlite/binding/ffi.py

```diff
@@ -151,12 +152,31 @@ def __call__(self, *args, **kwargs):
             return self._cfn(*args, **kwargs)
 
 
+def _importlib_resources_path_repl(package, resource):
+    """Replacement implementation of `import.resources.path` to avoid
+    deprecation warning following code at importlib_resources/_legacy.py
+    as suggested by https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy
+
+    Notes on differences from importlib.resources implementation:
+
+    The `_common.normalize_path(resource)` call is skipped because it is an
+    internal API and it is unnecessary for the use here. What it does is
+    ensuring `resource` is a str and that it does not contain path separators.
+    """ # noqa E501
+    return _impres.as_file(_impres.files(package) / resource)
+
+
+_importlib_resources_path = (_importlib_resources_path_repl
+                             if sys.version_info[:2] >= (3, 9)
+                             else _impres.path)
+
+
 _lib_name = get_library_name()
 
 
 pkgname = ".".join(__name__.split(".")[0:-1])
 try:
-    _lib_handle = importlib.resources.path(pkgname, _lib_name)
+    _lib_handle = _importlib_resources_path(pkgname, _lib_name)
```

Comment:
Once llvmlite bumps Python min version to 3.9, the code above can be safely replaced by:

```python
from importlib.resources import as_file, files
_lib_handle = as_file(files(pkgname).joinpath(_lib_name))
```